Provider/PhpcrMenuProvider.php

```diff
@@ -157,116 +158,81 @@ public function setRequest(Request $request = null)
 
-        $menuItem = $this->loader->load($menu);
         if (empty($menuItem)) {
-            throw new \InvalidArgumentException("Menu at '$name' is misconfigured (f.e. the route might be incorrect) and could therefore not be instanciated");
+            throw new \InvalidArgumentException(sprintf(
+                'Menu "%s" is misconfigured (f.e. the route might be incorrect) and could therefore not be instansiated',
+                $name
+            ));
         }
 
         return $menuItem;
     }
 
     /**
-     * Check if a menu node exists.
-     *
-     * If this method returns true, it means that you can call get() without
-     * an exception.
-     *
-     * @param string $name    Name of the menu to load. This can be an
-     *                        absolute PHPCR path or one relative to the menu root.
-     * @param array  $options
-     *
-     * @return bool Whether a menu with this name can be loaded by this provider.
+     * {@inheritDoc}
      */
     public function has($name, array $options = array())
     {
-        return $this->find($name, $options, false) instanceof NodeInterface;
+        $document = $this->find($name, $options);
+
+        if (null === $document) {
+            return false;
+        }
+
+        return true;
     }
 
     /**
-     * @param string $name    Name of the menu to load
+     * Find the named menu or `null` if the menu cannot be located.
+     *
+     * @param string $name Name of the menu to load
      * @param array  $options
-     * @param bool   $throw   Whether to throw an exception if the menu is not
-     *                        found or no valid menu. Returns false if $throw is false and there
-     *                        is no menu at $name.
      *
-     * @return object|bool The menu root found with $name or false if $throw
-     *                     is false and the menu was not found.
+     * @return ItemInterface|null
      *
-     * @throws \InvalidArgumentException Only if $throw is true throws this
-     *                                   exception if the name is empty or no menu found.
+     * @throws \RuntimeException If the found node does not implement the
+     *     correct interface.
      */
-    protected function find($name, array $options, $throw)
+    protected function find($name, array $options)
     {
-        if (empty($name)) {
-            if ($throw) {
-                throw new \InvalidArgumentException('The menu name may not be empty');
-            }
-
-            return false;
-        }
-
-        $dm = $this->getObjectManager();
-        $session = $dm->getPhpcrSession();
+        $manager = $this->getObjectManager();
+        $session = $manager->getPhpcrSession();
+        $path = PathHelper::absolutizePath($name, $this->getMenuRoot());
 
-        try {
-            $path = PathHelper::absolutizePath($name, $this->getMenuRoot());
-            PathHelper::assertValidAbsolutePath($path, false, true, $session->getNamespacePrefixes());
```

Comment:
@dbu I probably need to add this back .. do you remember what the purpose of this was?